scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -658,39 +664,166 @@ def test_as_euler_degenerate_symmetric_axes():
         [15, 0, 60],
         [35, 0, 75],
         [60, 180, 35],
-        [15, -180, 25],
-        ])
+        [15, -180, 25]])
 
     with pytest.warns(UserWarning, match="Gimbal lock"):
-        for axis1 in ['x', 'y', 'z']:
-            for axis2 in ['x', 'y', 'z']:
-                if axis1 == axis2:
-                    continue
+        for seq_tuple in permutations('xyz'):
+            # Extrinsic rotations
+            seq = ''.join([seq_tuple[0], seq_tuple[1], seq_tuple[0]])
+            rotation = Rotation.from_euler(seq, angles, degrees=True)
+            mat_expected = rotation.as_matrix()
+
+            angle_estimates = rotation.as_euler(seq, degrees=True)
+            mat_estimated = Rotation.from_euler(
+                seq, angle_estimates, degrees=True
+                ).as_matrix()
 
-                # Extrinsic rotations
-                seq = axis1 + axis2 + axis1
-                rotation = Rotation.from_euler(seq, angles, degrees=True)
-                mat_expected = rotation.as_matrix()
+            assert_array_almost_equal(mat_expected, mat_estimated)
+
+            # Intrinsic rotations
+            seq = seq.upper()
+            rotation = Rotation.from_euler(seq, angles, degrees=True)
+            mat_expected = rotation.as_matrix()
 
-                angle_estimates = rotation.as_euler(seq, degrees=True)
-                mat_estimated = Rotation.from_euler(
-                    seq, angle_estimates, degrees=True
-                    ).as_matrix()
+            angle_estimates = rotation.as_euler(seq, degrees=True)
+            mat_estimated = Rotation.from_euler(
+                seq, angle_estimates, degrees=True
+                ).as_matrix()
 
-                assert_array_almost_equal(mat_expected, mat_estimated)
+            assert_array_almost_equal(mat_expected, mat_estimated)
 
-                # Intrinsic rotations
-                seq = seq.upper()
-                rotation = Rotation.from_euler(seq, angles, degrees=True)
-                mat_expected = rotation.as_matrix()
 
-                angle_estimates = rotation.as_euler(seq, degrees=True)
-                mat_estimated = Rotation.from_euler(
-                    seq, angle_estimates, degrees=True
-                    ).as_matrix()
+def test_as_euler_test_algorithms():
```

Comment:
Don't you think it would be better to merge this test with `test_as_euler_asymmetric_axes` (and similar for the second one)? 

`test` helper can be renamed to `test_stats` (or something similar) and called along with `assert_allclose`.

I just don't see a very good reason to have these tests separate.

Reply:
Merged as you requested.